models/RepPoints/builder.py

```diff
@@ -0,0 +1,722 @@
+            pts_inits_concat,
+            transform,
+            y_first=False,
+            moment_transfer=self.moment_transfer
+        )
+        bboxes_inits_concat = X.reshape(bboxes_inits_concat_, (-4, batch_image, -1, -2))
+        normalize_term = X.concat(
+            [bboxes_out_strides["stride%s" % s] for s in stride], axis=1, name="normalize_term"
+        ) * scale
+        pts_init_loss = X.smooth_l1(
+            data=(bboxes_inits_concat - points_gts_init) / normalize_term,
+            scalar=3.0,
+            name="pts_init_l1_loss"
+        )
+        pts_init_loss = pts_init_loss * points_weight_init
+        pts_init_loss = X.bbox_norm(
+            data=pts_init_loss,
+            label=points_labels_init,
+            name="pts_init_norm_loss"
+        )
+        pts_init_loss = X.make_loss(
+            data=pts_init_loss,
+            grad_scale=0.5,
+            name="pts_init_loss"
+        )
+        points_init_labels = X.block_grad(points_labels_refine, name="points_init_labels")
+
+        # refine box branch
+        pts_refines_concat_ = X.concat(
+            [pts_coordinate_preds_refines["stride%s" % s] for s in stride],
+            axis=1,
+            name="pts_refines_concat_"
+        )
+        pts_refines_concat = X.reshape(pts_refines_concat_, (-3, -2), name="pts_refines_concat")
+        bboxes_refines_concat_ = _points2bbox(
+            mx.symbol,
+            pts_refines_concat,
+            transform,
+            y_first=False,
+            moment_transfer=self.moment_transfer
+        )
+        bboxes_refines_concat = X.reshape(bboxes_refines_concat_, (-4, batch_image, -1, -2))
+        pts_refine_loss = X.smooth_l1(
+            data=(bboxes_refines_concat - points_gts_refine) / normalize_term,
+            scalar=3.0,
+            name="pts_refine_l1_loss"
+        )
+        pts_refine_loss = pts_refine_loss * points_weight_refine
+        pts_refine_loss = X.bbox_norm(
+            data=pts_refine_loss,
+            label=points_labels_refine,
+            name="pts_refine_norm_loss"
+        )
+        pts_refine_loss = X.make_loss(
+            data=pts_refine_loss,
+            grad_scale=1.0,
+            name="pts_refine_loss"
+        )
+        points_refine_labels = X.block_grad(points_labels_refine, name="point_refine_labels")
+
+        return cls_loss, pts_init_loss, pts_refine_loss, points_init_labels, points_refine_labels
+
+    def get_prediction(self, conv_feat, im_info):
+        from models.RepPoints.point_ops import _gen_points, _points2bbox
+        p = self.p
+        batch_image = p.batch_image
+        stride = p.point_generate.stride
+        transform = p.point_generate.transform
+        pre_nms_top_n = p.proposal.pre_nms_top_n
+
+        pts_out_inits, pts_out_refines, cls_outs = self.get_output(conv_feat)
+
+        cls_score_dict = dict()
+        bbox_xyxy_dict = dict()
+        for s in stride:
+            pre_nms_top_n_ = pre_nms_top_n if s <= 32 else -1
```

Comment:
Please indicate this is a hack in the comment since if the image size goes down this could break.

Reply:
done.